sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/TracerProvider.java

```diff
@@ -0,0 +1,209 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.cosmos.implementation;
+
+import com.azure.core.util.Context;
+import com.azure.core.util.FluxUtil;
+import com.azure.core.util.logging.ClientLogger;
+import com.azure.core.util.tracing.ProcessKind;
+import com.azure.core.util.tracing.Tracer;
+import com.azure.cosmos.CosmosClientException;
+import com.azure.cosmos.models.CosmosAsyncItemResponse;
+import com.azure.cosmos.models.CosmosResponse;
+import reactor.core.publisher.Mono;
+import reactor.core.publisher.Signal;
+
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.concurrent.atomic.AtomicReference;
+
+public class TracerProvider {
+    private final ClientLogger logger = new ClientLogger(TracerProvider.class);
+    private final List<Tracer> tracers = new ArrayList<>();
+    public final static String DB_TYPE_VALUE = "cosmosdb";
+    public final static String DB_TYPE = "db.type";
+    public final static String DB_INSTANCE = "db.instance";
+    public final static String DB_URL = "db.url";
+    public static final String DB_STATEMENT = "db.statement";
+    public static final String ERROR_MSG = "error.msg";
+    public static final String ERROR_TYPE = "error.type";
+    public static final String ERROR_STACK = "error.stack";
+    public static final String MASTER_CALL = "masterCall";
+    public static final String NESTED_CALL = "nestedCall";
+
+    public static final Object ATTRIBUTE_MAP = "span-attributes";
+
+
+    public TracerProvider(Iterable<Tracer> tracers) {
+        Objects.requireNonNull(tracers, "'tracers' cannot be null.");
+        tracers.forEach(e -> this.tracers.add(e));
+    }
+
+    public boolean isEnabled() {
+        return tracers.size() > 0;
+    }
+
+    /**
+     * For each tracer plugged into the SDK a new tracing span is created.
+     * <p>
+     * The {@code context} will be checked for containing information about a parent span. If a parent span is found the
+     * new span will be added as a child, otherwise the span will be created and added to the context and any downstream
+     * start calls will use the created span as the parent.
+     *
+     * @param context Additional metadata that is passed through the call stack.
+     * @return An updated context object.
+     */
+    public Context startSpan(String methodName, Context context, ProcessKind processKind) {
+        Context local = Objects.requireNonNull(context, "'context' cannot be null.");
+        Objects.requireNonNull(processKind, "'processKind' cannot be null.");
+
+        for (Tracer tracer : tracers) {
+            local = tracer.start(methodName, local, processKind);
```

Comment:
Given the requirements to set these specific attributes on every span it would help if we could use 
```java
local = tracer.start(methodName, local); // will start the span
tracer.setAttribute(DB_INSTANCE, "myDatabaseName", local); // use the current span to set attributes on it
```

This way we don't have to add a new ProcessKind and [this](https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/core/azure-core-tracing-opentelemetry/src/main/java/com/azure/core/tracing/opentelemetry/OpenTelemetryTracer.java#L45) `startSpan` overload can be used as it.

Reply:
Removed ProcessKind dependency.